src/components/charts/helper.js

```diff
@@ -47,29 +47,43 @@ const getCategoryColorFrom2Colors = index => colorsForCharts.categorical2Colors[
 
 const getDataKey = showPercentage => (showPercentage ? 'percentage' : 'value');
 
-const transformArrangerDataToChart = (field) => {
+const transformArrangerDataToChart = (field, sqonValues) => {
   const chartData = [];
-  field.buckets.map(bucket =>
-    chartData.push({
-      name: bucket.key,
-      value: bucket.doc_count,
-    }),
-  );
+  field.buckets
+    .filter(bucket => (sqonValues === null || sqonValues.includes(bucket.key)))
+    .forEach(bucket =>
+      chartData.push({
+        name: bucket.key,
+        value: bucket.doc_count,
+      }),
+    );
   return chartData;
 };
 
-const transformArrangerDataToSummary = (field, chartType, title) => ({
+const transformArrangerDataToSummary = (field, chartType, title, sqonValues) => ({
   type: chartType,
   title,
-  data: transformArrangerDataToChart(field),
+  data: transformArrangerDataToChart(field, sqonValues),
 });
 
-const transformDataToCount = (field, label) => ({
+const transformDataToCount = (field, label, sqonValues) => ({
   label,
-  value: field.buckets.length,
+  value: sqonValues ? Math.min(field.buckets.length, sqonValues.length) : field.buckets.length,
 });
 
-const getCharts = (data, arrangerConfig) => {
+/**
+ * Return an array of selected values in a given field
+ * If no value selected, return null
+ */
+const getSQONValues = (sqon, field) => {
+  if (!sqon || !sqon.content) return null;
```

Comment:
you could also return an empty array right? that might be easier so you wouldn't have to check for null values, but I'll leave that up to you, just a suggestion

Reply:
well if it were an empty array it would just be `sqonValues.includes(bucket.key)`, I don't think you would need the OR statement. but again up to you